xml_converter/src/packaging_protobin.cpp

```diff
@@ -154,7 +161,16 @@ void proto_post_processing(ProtoWriterState* state, waypoint::Waypoint* proto) {
 
         for (size_t i = 1; i < state->textures.size(); i++) {
             waypoint::TextureData* texture_data = proto->add_textures();
-            texture_data->set_filepath(state->textures[i]->filename);
+            const Image* image = state->textures[i];
+            texture_data->set_filepath(image->filename);
+            if (fs::exists(fs::path(state->textures[i]->original_filepath))) {
+                fs::path output_path = fs::path(state->proto_directory) / image->filename;
+                fs::create_directories(output_path.parent_path());
+                fs::copy_file(fs::path(image->original_filepath), output_path, fs::copy_options::overwrite_existing);
+            }
+            else {
+                cout << "Warning: File path " << state->textures[i]->original_filepath << " not found." << endl;
+            }
```

Comment:
As discussed offline, we want to move this file copy into image_to_proto's else statement which will have it only get executed the same number of times as this, but be in the same file as the xml image copy code.